libkbfs/keybase_service_base.go

```diff
@@ -405,16 +415,61 @@ func (k *KeybaseServiceBase) LoadUserPlusKeys(ctx context.Context,
 // KeybaseServiceBase.
 func (k *KeybaseServiceBase) LoadTeamPlusKeys(
 	ctx context.Context, tid keybase1.TeamID) (TeamInfo, error) {
-	panic("The Keybase service doesn't support LoadTeamPlusKeys yet")
+	// No caching until the invalidations are ready.
+
+	arg := keybase1.LoadTeamPlusApplicationKeysArg{
+		Id:          tid,
+		Application: keybase1.TeamApplication_KBFS,
+	}
+	res, err := k.teamsClient.LoadTeamPlusApplicationKeys(ctx, arg)
+	if err != nil {
+		return TeamInfo{}, err
+	}
+
+	if tid != res.Id {
+		return TeamInfo{}, fmt.Errorf(
+			"TID doesn't match: %s vs %s", tid, res.Id)
+	}
+
+	info := TeamInfo{
+		Name:      libkb.NormalizedUsername(res.Name),
+		TID:       res.Id,
+		CryptKeys: make(map[KeyGen]kbfscrypto.TLFCryptKey),
+		Writers:   make(map[keybase1.UID]bool),
+		Readers:   make(map[keybase1.UID]bool),
+	}
+	for _, key := range res.ApplicationKeys {
+		info.CryptKeys[KeyGen(key.KeyGeneration)] =
+			kbfscrypto.MakeTLFCryptKey(key.Key)
+	}
+
+	for _, user := range res.Writers {
+		info.Writers[user.Uid] = true
+	}
+	for _, user := range res.OnlyReaders {
+		info.Readers[user.Uid] = true
+	}
+	return info, nil
 }
 
 // GetCurrentMerkleSeqNo implements the KeybaseService interface for
 // KeybaseServiceBase.
 func (k *KeybaseServiceBase) GetCurrentMerkleSeqNo(ctx context.Context) (
 	MerkleSeqNo, error) {
-	// TODO: call into the service once they implement a way to get
-	// the current seqno.
-	return UnknownMerkleSeqNo, nil
+	const merkleFreshness = int(time.Second * 60 / time.Millisecond)
+	res, err := k.merkleClient.GetCurrentMerkleRoot(ctx, merkleFreshness)
```

Comment:
Does this guarantee that anything returned with no error is from the past minute?